src/lib/core/components/ScatterplotRsquareTable.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  XYPlotDataSeries,
+  XYPlotData,
+} from '@veupathdb/components/lib/types/plots';
+import { VariableTreeNode } from '../types/study';
+
+interface facetDataProps {
+  label: string;
+  data?: XYPlotData | undefined;
+}
+
+export interface Props {
+  data: XYPlotDataSeries[] | facetDataProps[] | undefined;
+  isFaceted: boolean;
+  overlayVariable?: VariableTreeNode;
+  facetVariable?: VariableTreeNode;
+}
+
+export function ScatterplotRsquareTable({
+  data,
+  isFaceted,
+  overlayVariable,
+  facetVariable,
+}: Props) {
+  // non-facet or facet data
+  const filteredData = !isFaceted
+    ? overlayVariable != null
+      ? (data as XYPlotDataSeries[])?.filter((data) =>
```

Comment:
I have a few ideas for getting rid of these type assertions. Because my ideas partially depend on how you decide to resolve the TypeScript error I pointed out, I'll wait until you've addressed the TypeScript error, and then explain more.